pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -532,12 +532,19 @@ func (e *Controller) syncService(ctx context.Context, key string) error {
 	}
 
 	logger.V(4).Info("Update endpoints", "service", klog.KObj(service), "readyEndpoints", totalReadyEps, "notreadyEndpoints", totalNotReadyEps)
+	shouldFlagRVAsStale := false
 	if createEndpoints {
 		// No previous endpoints, create them
 		_, err = e.client.CoreV1().Endpoints(service.Namespace).Create(ctx, newEndpoints, metav1.CreateOptions{})
 	} else {
 		// Pre-existing
-		_, err = e.client.CoreV1().Endpoints(service.Namespace).Update(ctx, newEndpoints, metav1.UpdateOptions{})
+		var updatedEndpoints *v1.Endpoints
+		updatedEndpoints, err = e.client.CoreV1().Endpoints(service.Namespace).Update(ctx, newEndpoints, metav1.UpdateOptions{})
+		// an webhook could potentially cause endpoints update operation became noop and return same resourceVersion.
+		// see https://github.com/kubernetes/kubernetes/issues/126578#issuecomment-2314368183
+		if updatedEndpoints != nil && updatedEndpoints.ResourceVersion != currentEndpoints.ResourceVersion {
```

Comment:
`Update` returns an empty object instead of `nil` when it fails, so `updatedEndpoints.ResourceVersion` would be empty and different from `currentEndpoints.ResourceVersion`, causing `shouldFlagRVAsStale` to be `true`. Even though the check at L549 would prevent `e.staleEndpointsTracker.Stale(currentEndpoints)` from being executed, having an incorrect flag leftover may cause other issues in the future. I think we should check `if err == nil && updatedEndpoints.ResourceVersion != currentEndpoints.ResourceVersion`

Reply:
make sense, i can adjust it to make the stale RV logic more explicitly and closer to the `update branch`.
The current way to avoid major changes to existing code.

Reply:
it avoid major changes but it has now different code paths in nested `if` statements, maybe there is not better way, but see if you can find another